crates/transaction-pool/src/maintain.rs

```diff
@@ -64,17 +95,100 @@ where
     // keeps track of the state of the pool wrt to blocks
     let mut maintained_state = MaintainedPoolState::InSync;
 
+    // the future that reloads accounts from state
+    let mut reload_accounts_fut = Fuse::terminated();
+
+    // The update loop that waits for new blocks and reorgs and performs pool updated
     // Listen for new chain events and derive the update action for the pool
     loop {
+        trace!(target = "txpool", state=?maintained_state, "awaiting new block or reorg");
+
         metrics.set_dirty_accounts_len(dirty_addresses.len());
+        let pool_info = pool.block_info();
 
-        let Some(event) = events.next().await else { break };
+        // after performing a pool update after a new block we have some time to properly update
+        // dirty accounts and correct if the pool drifted from current state, for example after
+        // restart or a pipeline run
```

Comment:
why after a restart? we dont save the txpool state anywhere, so when would it affect?

Reply:
because we only get events after pipeline finished 
will clarify this